src/modules/csm/csm.py

```diff
@@ -93,47 +98,38 @@ def execute_module(self, last_finalized_blockstamp: BlockStamp) -> ModuleExecute
     def build_report(self, blockstamp: ReferenceBlockStamp) -> tuple:
         self.validate_state(blockstamp)
 
-        prev_root = self.w3.csm.get_csm_tree_root(blockstamp)
-        prev_cid = self.w3.csm.get_csm_tree_cid(blockstamp)
+        last_report = self._get_last_report(blockstamp)
+        rewards_tree_root, rewards_cid = last_report.rewards_tree_root, last_report.rewards_tree_cid
+        strikes_tree_root, strikes_cid = last_report.strikes_tree_root, last_report.strikes_tree_cid
 
-        if (prev_cid is None) != (prev_root == ZERO_HASH):
-            raise InconsistentData(f"Got inconsistent previous tree data: {prev_root=} {prev_cid=}")
+        (
+            total_distributed,
+            total_rewards,
+            strikes,
+            logs,
+        ) = self.calculate_distribution(blockstamp, last_report)
 
-        total_distributed, total_rewards, logs = self.calculate_distribution(blockstamp)
+        if total_distributed:
+            rewards_tree = self.make_rewards_tree(total_rewards)
+            rewards_tree_root = rewards_tree.root
+            rewards_cid = self.publish_tree(rewards_tree)
 
-        if total_distributed != sum(total_rewards.values()):
-            raise InconsistentData(f"Invalid distribution: {sum(total_rewards.values())=} != {total_distributed=}")
+        if strikes:
```

Comment:
If no current strikes, but there were strikes before (due to shifting), we need to pass zero hash to wipe it in the contract to not have the tree consists of leafs with `[0,0,0,0,0]` because it doesn't make sense.

If we pass the same root as previous it can be interpreted as a result of non-shifted strikes. I'd like to avoid it.